pkg/ruler/manager.go

```diff
@@ -331,6 +331,12 @@ func (*DefaultMultiTenantManager) ValidateRuleGroup(g rulefmt.RuleGroup) []error
 		return errs
 	}
 
+	if !r.cfg.TenantFederation.Enabled && len(g.SourceTenants) > 0 {
+		errs = append(errs, fmt.Errorf("invalid rules config: rule group '%s' is a federated rule group, but rules federation is disabled. "+
+			"To enable the feature, configure the Mimir or GEM ruler instance with the following parameter or contact your service administrator: "+
+			"set -"+TenantFederationFlag+"=true as a CLI argument, ruler.tenant_federation.enabled: true in YAML", g.Name))
```

Comment:
@osg-grafana can you please review this error message? The user will see this when they try to use `mimirtool` for a feature which is disabled in the Mimir cluster. They or their admin should enable that feature on the Mimir cluster before the user can use `mimirtool`

Reply:
nit: I'm not sure we should give this much information about the system - not that it's a security hole or something, but that it feels way too verbose to be useful.

My suggestion is to leave it at:

```
invalid rules config: rule group '%s' is a federated rule group, but rules federation is disabled. Please contact your service administrator to have it enabled.
```